src/core/simulator/entity/floor_entity.cpp

```diff
@@ -133,7 +133,9 @@ namespace argos {
 
 #ifdef ARGOS_WITH_FREEIMAGE
       virtual void SaveAsImage(const std::string& str_path) {
-         fipImage cImage(FIT_BITMAP, m_unPixelsPerMeter * m_cHalfArenaSize.GetX()*2, m_unPixelsPerMeter * m_cHalfArenaSize.GetY()*2, 24);
+         fipImage cImage(FIT_BITMAP,
+                         static_cast<UInt32>(m_unPixelsPerMeter * m_cHalfArenaSize.GetX()*2),
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a implicit float -> int conversion warning without the cast